pgx/src/lib.rs

```diff
@@ -245,6 +245,21 @@ macro_rules! pg_magic_func {
                 float8byval: pgx::pg_sys::USE_FLOAT8_BYVAL as i32,
             };
 
+            #[cfg(any(feature = "pg15"))]
+            const MY_MAGIC: pgx::pg_sys::Pg_magic_struct = pgx::pg_sys::Pg_magic_struct {
+                len: size_of::<pgx::pg_sys::Pg_magic_struct>() as i32,
+                version: pgx::pg_sys::PG_VERSION_NUM as i32 / 100,
+                funcmaxargs: pgx::pg_sys::FUNC_MAX_ARGS as i32,
+                indexmaxkeys: pgx::pg_sys::INDEX_MAX_KEYS as i32,
+                namedatalen: pgx::pg_sys::NAMEDATALEN as i32,
+                float8byval: pgx::pg_sys::USE_FLOAT8_BYVAL as i32,
+                abi_extra: [
+                    b'P' as i8, b'o' as i8, b's' as i8, b't' as i8, b'g' as i8, b'r' as i8,
+                    b'e' as i8, b'S' as i8, b'Q' as i8, b'L' as i8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
```rust
unsafe { std::mem::transmute(b"PostgreSQL\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0") };
```
would be nicer, but [is technically unsound](https://users.rust-lang.org/t/how-to-convert-i8-to-u8/16308/11?u=smitop)

Reply:
If we're seriously going to quibble over this, then I would prefer my own flavor of incomprehensible conversion nonsense:
```suggestion
{
    // array::from_fn isn't const yet, boohoo, so const-copy a bstr
    let magic = b"PostgreSQL";
    let mut abi = [0 as libc::c_char; 32];
    let mut i = 0;
    while i < magic.len() {
        abi[i] = magic[i] as _;
        i += 1;
    }
    abi
}
```

At least then we can read the string.

Reply:
I changed it to use your implementation, using `i8` instead of `libc::c_char` since `libc` isn't always available where the macro is expanded.